content/en/docs/guides/retry.md

```diff
@@ -0,0 +1,129 @@
+
+The graph below shows architectural overview of gRPC retry internal.
+
+
+```mermaid
+sequenceDiagram
+  Application ->> gRPC Client: Configure retry policy. <br> Send request to dns:///my-service
+  gRPC Client ->> Server: Create retry attempt 1 
+  Server -->> gRPC Client : RPC closed with error 
+  gRPC Client ->> Server: Create retry attempt 2 
+  Server -->> gRPC Client: Successful 
+  gRPC Client ->> Application: No more retry. Proceed.
+```
+
+### Retry configuration
+Retries are enabled by default even without specifying a retry policy. You can also disable retries entirely.
+
+You can optimize your application's retry functionality by focusing on key steps and configurations that gRPC supports.
+* Max number of retry attempts
+* Exponential backoff 
+* Set of retryable status codes
+
+Retry is configurable via [gRPC Service Config], at a per-method granularity.
+  The configuration contains the following knobs:
+
+```
+"retryPolicy": {
+  "maxAttempts": 4,
+  "initialBackoff": "0.1s",
+  "maxBackoff": "1s",
+  "backoffMultiplier": 2,
+  "retryableStatusCodes": [
+    "UNAVAILABLE"
+  ]
+}
+```
+
+The actual initial backoff delay is a random time period within `initialBackoff` value configured above (between 0-100ms).
+
+gRPC supports throttle limit that prevents server overload due to retries. Below is an examples of retry throttle configuration:
+
+```
+"retryThrottling": {
+  "maxTokens": 10,
+  "tokenRatio": 0.1
+}
+```
+
+For each server, the gRPC client tracks a `token_count` (initially set to `maxTokens`). Failed RPCs decrement the count by 1, successful RPCs increment it by  `tokenRatio`.  If the `token_count` falls below half of `maxTokens`, retries are paused until the count recovers.
+
+Further, hedging is a complementary feature to retries and can be configured similarly. For more details, see the [hedging guide].
+
+### Retry Observability
+
+gRPC supports exposing OpenCensus and OpenTelemetry metrics when retry functionality is enabled. Here's an example of the OpenTelemetry retry attempt statistics available:
+* `grpc.client.attempt.started`
+* `grpc.client.attempt.duration`
+* `grpc.client.attempt.sent_total_compressed_message_size`
+* `grpc.client.attempt.rcvd_total_compressed_message_size`
+
+Metrics at per cal level:
+* `grpc.client.call.duration`
+
+And server side metrics:
+* `grpc.server.call.started`
+* `grpc.server.call.sent_total_compressed_message_size`
+* `grpc.server.call.rcvd_total_compressed_message_size`
+* `grpc.server.call.duration`
+
+Find in-depth metrics and tracing information, along with configuration instructions, in the [gRFC for Otel metrics], [gRFC for retry status].
+
+
+{{% alert title="Transparent Retry" color="info" %}}
+Failure can occur in different stages. Even without an explicit retry policy, gRPC may perform transparent retries. The extent of these retries depends on when the failure happens:
```

Comment:
This should be moved above the "observability" section. But I'm thinking it might be better integrated into a paragraph. This level of detail is probably not helpful to most users, and saying "use caution with this type of retry" is weird because the user has no control over it.